lib/segment/src/problems/unindexed_field.rs

```diff
@@ -236,14 +236,14 @@ fn infer_schema_from_field_condition(field_condition: &FieldCondition) -> Vec<Pa
     inferred
 }
 
-struct Extractor<'a> {
+pub struct Extractor<'a> {
     payload_schema: &'a HashMap<PayloadKeyType, PayloadFieldSchema>,
```

Comment:
I'm not completely happy with the way we're reusing `Extractor` here but I couldn't come up with a better idea.
If you have a suggestion on how to improve this, please let me know!

The advantages are, we're automatically testing `Extractor` too and don't have any code duplication.

Reply:
#4945